### PR TITLE
Fix false positives/negatives in selector-pseudo-class-no-unknown

### LINKED ISSUE #5173: Fix false negatives for :contains in selector-pseudo-class-no-unknown
Ref: https://github.com/stylelint/stylelint/pull/5171#issue-580426996

### PR DESCRIPTION
**Pull request by Mouvedia**
> Which issue, if any, is this issue related to?

Closes #5173

> Is there anything in the PR that needs further explanation?

check the commit message for details
prompted by https://github.com/stylelint/stylelint/issues/5205#issuecomment-1060017652

### CODE REVIEW

File: lib/reference/keywordSets.js [hunk 1]
```diff
@@ -277,10 +274,21 @@ keywordSets.otherPseudoClasses = new Set([
 	'target',
 	'user-error',
```
Comment (Mouvedia) [inline lib/reference/keywordSets.js:275]: not sure if it should be removed or not
see https://lists.w3.org/Archives/Public/www-style/2015Sep/0111.html

Comment (jeddy3) [reply lib/reference/keywordSets.js:275]: If it's been dropped, let's remove it too.

**Review by jeddy3 — CHANGES_REQUESTED** [review]
Thanks!

Let's remove `user-error` too.

**Comment by Mouvedia** [comment]
> Let's remove `user-error` too.

done

**Review by jeddy3 — APPROVED** [review]
Thank you! LGTM.

File: lib/reference/keywordSets.js [hunk 2]
```diff
@@ -256,7 +254,6 @@ keywordSets.otherPseudoClasses = new Set([
 	'hover',
 	'indeterminate',
 	'in-range',
-	'invalid',
```
Comment (ybiquitous) [inline lib/reference/keywordSets.js:259]: [trivial] Alphabetical order?

**Review by ybiquitous — APPROVED** [review]
Looks good! 👍🏼

File: lib/reference/keywordSets.js (hunk 2 above)
Comment (ybiquitous) [reply lib/reference/keywordSets.js:259]: I understand it. 👍🏼

**Comment by jeddy3** [comment]
Changelog entry added:

> - Fixed: `selector-pseudo-class-no-unknown` false positives and negatives ([#5956](https://github.com/stylelint/stylelint/pull/5956)).

**Pull request merged**